app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -297,6 +285,9 @@ const SiteHeaderComponent = MountWidget.extend(
       this._mousetrap.reset();
 
       document.removeEventListener("click", this._dismissFirstNotification);
+
+      const dockAnchor = document.querySelector(".header-dock-anchor");
+      this._stickyHeaderDockObserver?.unobserve(dockAnchor);
```

Comment:
Im curious what happens when we pass undefined to unobserve ?

Reply:
That won't happen because `this._stickyHeaderDockObserver` is null unless we have `dockAnchor` 

https://github.com/discourse/discourse/pull/13840/files/3e32846b4ac96c482352cad22d5d378f8c03709e#diff-c5865c5d26ff1eef7ee0e664a9eec8ca10c045c828aa90a5d47bc7def5381dadR256

Reply:
Yes that was more a general question I had about this, but I will try myself :p